src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2087,8 +2108,12 @@ mgp_error mgp_graph_delete_vertex(struct mgp_graph *graph, mgp_vertex *vertex) {
           throw SerializationException{"Cannot serialize removing a vertex."};
       }
     }
-    if (graph->ctx->trigger_context_collector && *result) {
-      graph->ctx->trigger_context_collector->RegisterDeletedObject(**result);
+    auto &ctx = graph->ctx;
+
+    ctx->execution_stats[query::ExecutionStats::Key::DELETED_NODES] += 1;
```

Comment:
I would put this into the inner body of the following `if` statement, because result can be `std::nullopt` here, which means no deletion.